utils/update_checkout/update_checkout/update_checkout.py

```diff
@@ -100,20 +115,25 @@ def get_branch_for_repo(config, repo_name, scheme_name, scheme_map,
     cross_repo = False
     repo_branch = scheme_name
     if scheme_map:
-        scheme_branch = scheme_map[repo_name]
-        repo_branch = scheme_branch
-        remote_repo_id = config['repos'][repo_name]['remote']['id']
-        if remote_repo_id in cross_repos_pr:
-            cross_repo = True
-            pr_id = cross_repos_pr[remote_repo_id]
-            repo_branch = "ci_pr_{0}".format(pr_id)
-            shell.run(["git", "checkout", scheme_branch],
-                      echo=True)
-            shell.capture(["git", "branch", "-D", repo_branch],
-                          echo=True, allow_non_zero_exit=True)
-            shell.run(["git", "fetch", "origin",
-                       "pull/{0}/merge:{1}"
-                       .format(pr_id, repo_branch), "--tags"], echo=True)
+        try:
+            scheme_branch = scheme_map[repo_name]
+            repo_branch = scheme_branch
+            remote_repo_id = config['repos'][repo_name]['remote']['id']
+            if remote_repo_id in cross_repos_pr:
+                cross_repo = True
+                pr_id = cross_repos_pr[remote_repo_id]
+                repo_branch = "ci_pr_{0}".format(pr_id)
+                shell.run(["git", "checkout", scheme_branch],
+                          echo=True)
+                shell.capture(["git", "branch", "-D", repo_branch],
+                              echo=True, allow_non_zero_exit=True)
+                shell.run(["git", "fetch", "origin",
+                           "pull/{0}/merge:{1}"
+                           .format(pr_id, repo_branch), "--tags"], echo=True)
+        except KeyError as e:
+            print(f"Failed to look up {repo_name} in scheme", file=sys.stderr)
+            raise e
```

Comment:
Do you want `raise e` or just `raise`?
I believe using `raise e` will replace the call stack, and so might either be removing information, or just producing more verbose output if this ends up giving a chained error.

Reply:
Yeah, with the `raise e` its; 
```
Error on repo "/private/tmp/swift-test/swift-cmark-gfm": Traceback (most recent call last):
  File "/private/tmp/swift-test/swift/utils/update_checkout/update_checkout/update_checkout.py", line 156, in update_single_repository
    checkout_target, cross_repo = get_branch_for_repo(
  File "/private/tmp/swift-test/swift/utils/update_checkout/update_checkout/update_checkout.py", line 135, in get_branch_for_repo
   raise e
  File "/private/tmp/swift-test/swift/utils/update_checkout/update_checkout/update_checkout.py", line 119, in get_branch_for_repo
    scheme_branch = scheme_map[repo_name]
KeyError: 'swift-cmark-gfm'
```

Without it, it's just;
```
Error on repo "/private/tmp/swift-test/swift-cmark-gfm": Traceback (most recent call last):
  File "/private/tmp/swift-test/swift/utils/update_checkout/update_checkout/update_checkout.py", line 156, in update_single_repository
    checkout_target, cross_repo = get_branch_for_repo(
  File "/private/tmp/swift-test/swift/utils/update_checkout/update_checkout/update_checkout.py", line 119, in get_branch_for_repo
    scheme_branch = scheme_map[repo_name]
KeyError: 'swift-cmark-gfm'
```

Nice catch.